src/pipeline.py

```diff
@@ -266,9 +267,61 @@ def __init__(self, **kwargs):
         )
 
 
+class ONNX_Pipeline(Pipeline):
+    def __init__(
+        self,
+        *,
+        model_type: Optional[str] = None,
+        pretrained_config: Optional[str] = None,
+        pretrained_model: Optional[str] = None,
+        config_args: Dict[str, Any],
+        tokenizer: str,
+        device: torch.device,
+        dtype: torch.dtype,
+        fast_init: bool = False,
+        trust_remote_code: bool = False,
+    ):
+        self.global_metrics = {}
+        log_rank_n("*** Setting up tokenizer", logger.info)
+        t0 = time.perf_counter()
+        self.tokenizer = AutoTokenizer.from_pretrained(tokenizer)
+
+        self.tokenizer.add_special_tokens({"pad_token": "[PAD]"})
+        t1 = time.perf_counter()
+
+        self.device = device
+
+        if device == torch.device("cpu"):
+            provider = "CPUExecutionProvider"
+        elif device == torch.device("cuda"):
+            provider = "CUDAExecutionProvider"
+
+        self.dtype = dtype
+
+        if self.dtype != torch.float32:
```

Comment:
It still doesn't support float16?

Reply:
it does but not for santacoder. it requires a merged graph (at least if you compile via optimum)